tests/gnrc_tcp/tests/shared_func.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2018   Simon Brummer <simon.brummer@posteo.de>
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import os
+import re
+import pexpect
+import random
+
+def drainTextBuffer(child):
+    output = ''
+    try:
+        while True:
+            output += child.read_nonblocking(size = 1, timeout = 0.5)
+
+    except pexpect.TIMEOUT:
+        pass
+
+    return output
+
+def getHostTapDevice():
+    # Check if given tap device is part of a network bridge
+    # if so use bridged interface instead of given tap device
+    tap = os.environ["TAPDEV"]
+    result = os.popen('bridge link | grep ' + tap)
```

Comment:
> 
> The python doc [recommends](https://docs.python.org/3/library/subprocess.html#replacing-os-popen-os-popen2-os-popen3) replacing `os.popen` with `subprocess.Popen` (as in the end [the latter is used to implement the first](https://docs.python.org/3/library/os.html#os.popen).

Using subprocess.Popen instead of os.popen seems block further script execution. I would like to keep os.popen as I see no real benefit using subprocess.Popen instead.

Reply:
You can make it non-blocking, however I don't really understand why you want to use it unblocking, as you then read its full output below anyway.

Reply:
Of course it should block until completion. The problem that I have if the subprocess.Popen did unblock after half a minute and I don't know why.